### PR TITLE
Add OSType as a label for node constraints

### PR DESCRIPTION
**Pull request by wsong**

### CODE REVIEW

**Comment by wsong** [comment]
The reason why I made this change was because if you have Windows nodes in your Swarm cluster, `docker build` commands will sometimes attempt to build Linux-based images on Windows nodes. There doesn't appear to be an easy way to catch this ahead of time, but I've updated this PR to catch that specific error and recommend that users set the build arg constraints so that their builds run on the correct nodes.

**Comment by allencloud** [comment]
CI fails. PTAL @wsong ☀️

**Comment by wsong** [comment]
The failing test doesn't seem to be obviously related:

```
=== RUN   TestInitialize
--- PASS: TestInitialize (0.00s)
=== RUN   TestRegister
--- FAIL: TestRegister (35.27s)
	token_test.go:44: Timed out
```

Any idea what's going on here?

**Comment by dongluochen** [comment]
The docker hub discovery (token) service is down. It's tracked by https://github.com/docker/swarm/issues/2676.

**Comment by wsong** [comment]
Re-ran tests. Test failures appear to be unrelated.

**Comment by wsong** [comment]
Ping @nishanttotla and @dongluochen; can you guys take a look at this when possible? This change makes building on a cluster with both Windows and Linux nodes much easier.

**Comment by wsong** [comment]
We already have OperatingSystem, for what it's worth:

https://github.com/docker/swarm/blob/master/cluster/engine.go#L563

I'm happy to add any other labels you'd like to see.

File: cluster/engine_test.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ var (
 		Driver:          "driver-test",
 		KernelVersion:   "1.2.3",
 		OperatingSystem: "golang",
+		OSType:          "linux",
```
Comment (dongluochen) [inline cluster/engine_test.go]: Can you add a test case for the error case? That Swarm would not be able to build the target image.

Comment (wsong) [inline cluster/engine_test.go]: We don't really have any good test cases at the `api/handlers.go` level, as far as I can tell. Is there a specific place you hand in mind for this test?

Comment (dongluochen) [inline cluster/engine_test.go]: Since this is about build, maybe update `test/integration/api/build.bats`.

**Comment by wsong** [comment]
Hmm, I added an integration test, but it's not passing right now because when you download a Windows image on a Linux box, the error message you get back is "unknown blob", not the OS error. This appears to be an engine bug.

I've tested out this PR manually in a mixed Linux-Windows cluster, so maybe we can circle back to the integration test later? At the moment, I believe this test isn't going to work unless we can set up a Windows node in the test env.

**Comment by dongluochen** [comment]
We can skip this test by adding `skip`. Please add a swarm issue to track it, and link to an engine issue.

**Comment by wsong** [comment]
Done.

**Comment by dongluochen** [comment]
LGTM

ping @allencloud @nishanttotla

**Pull request merged**